Pulled in your branch on current master then ran:

```
bundle exec rake build:converter:opal
npm install                                                 # for the changed deps
npm run build
npm run test
```

It crashes with:
```
$ npm run test

> asciidoctor-reveal.js@1.1.0-dev test /home/olivier/src/asciidoc/asciidoctor-reveal.js
> node test/smoke.js


/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:4993
      throw exception;
      ^
NotImplementedError: String#<< not supported. Mutable String methods are not supported in Opal.
    at String.TMP_String_$lt$lt_1 [as $<<] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:20858:19)
    at $Document.TMP_55 (/home/olivier/src/asciidoc/asciidoctor-reveal.js/dist/main.js:1933:20)
    at $Document.$$instance_eval [as $instance_eval] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:3619:24)
    at Opal.send (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:1605:19)
    at $Converter.$$document [as $document] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/dist/main.js:1922:14)
    at $Converter.$$__send__ [as $__send__] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:3546:21)
    at Object.Opal.send (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:1605:19)
    at $Converter.alias [as $send] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/opal-runtime/src/opal.js:1779:19)
    at $Converter.$$convert [as $convert] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/dist/main.js:211:26)
    at $Document.$$convert [as $convert] (/home/olivier/src/asciidoc/asciidoctor-reveal.js/node_modules/asciidoctor.js/dist/asciidoctor.js:27996:35)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! asciidoctor-reveal.js@1.1.0-dev test: `node test/smoke.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the asciidoctor-reveal.js@1.1.0-dev test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/olivier/.npm/_logs/2017-10-10T21_33_31_209Z-debug.log
```

I'm trying from a clean directory using `npm i ../asciidoctor-reveal.js` and i'll report back. Still I expected smoke testing to work from inside the repo.

--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,7 @@ node_modules/
 npm-debug.log
 build/
 /dist/main.js
+/asciidoctor-reveal.js-*.tgz
 
 # ruby stuff
 /.bundle/

--- a/HACKING.adoc
+++ b/HACKING.adoc
@@ -195,31 +195,33 @@ You can apply that label to a pull request that is complete and ready for review
 
 Makes triaging easier.
 
-
 == Node package
 
 === Test a local asciidoctor-reveal.js version
 
-In order to test the Node package, you need to create a test project adjacent to the clone of the `asciidoctor-reveal.js` repository:
+In order to test the Node package, you first need to create a tarball of the project using the command `npm pack`.
+This command will produce a file named `asciidoctor-reveal.js-<version>.tgz` in the working directory.
+
+Then, create a test project adjacent to the clone of the [.path]_asciidoctor-reveal.js_ repository:
 
  $ mkdir test-project
  $ cd test-project
 
-Now, install the dependencies.
+Now, install the dependencies from the tarball:
+
+ $ npm i --save ../asciidoctor-reveal.js/asciidoctor-reveal.js-<version>.tgz
+
+NOTE: The relative portion of the last command is where you are installing the local `asciidoctor-reveal.js` version from.
+
+Then proceed as documented in the `README.adoc`.
+
+=== Upgrade Asciidoctor.js version
 
 WARNING: It is important to track `Asciidoctor.js` and `bestikk-opal-compiler` versions together.
 The `opal-compiler` used to compile our node package must match `asciidoctor.js` `opal-runtime`.
 The former is explicitly installed by users on install and the later is specified in our `package.json`.
 When you update one remember to update the other.
 Versions known to work together can be found here, just replace <tag> with the `asciidoctor.js` release you are interested in: https://github.com/asciidoctor/asciidoctor.js/blob/<tag>/package.json
-
- $ npm i --save asciidoctor.js@1.5.6-preview.3
- $ npm i --save ../asciidoctor-reveal.js
-
-NOTE: The relative portion of the last command is where you are installing the local `asciidoctor-reveal.js` version from.
-
-Then proceed as documented in the `README.adoc`.
-
 
 == RubyGem package
 

--- a/Rakefile
+++ b/Rakefile
@@ -9,65 +9,59 @@ require 'tilt'
 CONVERTER_FILE = 'lib/asciidoctor-revealjs/converter.rb'
 TEMPLATES_DIR = 'templates'
 
-namespace :build do
-  require 'asciidoctor-templates-compiler'
-  require 'slim-htag'
-
-  file CONVERTER_FILE, [:mode] => FileList["#{TEMPLATES_DIR}/*"] do |t, args|
-    #require 'asciidoctor-templates-compiler'
-    require_relative 'lib/asciidoctor-templates-compiler'
-    require 'slim-htag'
-
-    generator = if args[:mode] == :opal
-      Temple::Generators::ArrayBuffer.new(freeze_static: false)
-    else
-     Temple::Generators::StringBuffer
-    end
-
-    File.open(CONVERTER_FILE, 'w') do |file|
-      $stderr.puts "Generating #{file.path}."
-      Asciidoctor::TemplatesCompiler::RevealjsSlim.compile_converter(
-          templates_dir: TEMPLATES_DIR,
-          class_name: 'Asciidoctor::Revealjs::Converter',
-          register_for: ['revealjs'],
-          backend_info: {
-            basebackend: 'html',
-            outfilesuffix: '.html',
-            filetype: 'html',
-          },
-          delegate_backend: 'html5',
-          engine_opts: {
-            generator: generator,
-          },
-          pretty: (args[:mode] == :pretty),
-          output: file)
-    end
-  end
-
-  namespace :converter do
-    desc 'Compile Slim templates and generate converter.rb (pretty mode)'
-    task :pretty do
-      Rake::Task[CONVERTER_FILE].invoke(:pretty)
-    end
-
-    desc 'Compile Slim templates and generate converter.rb (fast mode)'
-    task :fast do
-      Rake::Task[CONVERTER_FILE].invoke
-    end
-
-    desc 'Compile Slim templates and generate converter.rb (opal mode)'
-    task :opal do
-      Rake::Task[CONVERTER_FILE].invoke(:opal)
-    end
-  end
-
-  task :converter => 'converter:pretty'
+file CONVERTER_FILE => FileList["#{TEMPLATES_DIR}/*"] do
+  build_converter :fast
 end
 
-task :build => 'build:converter:pretty'
+namespace :build do
+  desc 'Compile Slim templates and generate converter.rb'
+  task :converter => 'clean' do
+    build_converter
+  end
+
+  desc 'Compile Slim templates and generate converter.rb for Opal'
+  task 'converter:opal' => 'clean' do
+    build_converter :opal
+  end
+end
+
+task :build => 'build:converter'
 
 task :clean do
   rm_rf CONVERTER_FILE
+end
+
+def build_converter(mode = :pretty)
+  #require 'asciidoctor-templates-compiler'
+  require_relative 'lib/asciidoctor-templates-compiler'
+  require 'slim-htag'
+
+  generator = if mode == :opal
+    Temple::Generators::ArrayBuffer.new(freeze_static: false)
+  else
+    Temple::Generators::StringBuffer
+  end
+
+  File.open(CONVERTER_FILE, 'w') do |file|
+    puts "Generating #{file.path} (mode: #{mode})."
+
+    Asciidoctor::TemplatesCompiler::RevealjsSlim.compile_converter(
+      templates_dir: TEMPLATES_DIR,
+      class_name: 'Asciidoctor::Revealjs::Converter',
+      register_for: ['revealjs'],
+      backend_info: {
+        basebackend: 'html',
+        outfilesuffix: '.html',
+        filetype: 'html',
+      },
+      delegate_backend: 'html5',
+      engine_opts: {
+        generator: generator,
+      },
+      pretty: (mode == :pretty),
+      output: file
+    )
+  end
 end
 
 DocTest::RakeTasks.new do |t|
@@ -80,7 +74,7 @@ end
 task 'prepare-converter' do
   # Run as an external process to ensure that it will not affect tests
   # environment with extra loaded modules (especially slim).
-  `bundle exec rake build:converter:fast`
+  `bundle exec rake #{CONVERTER_FILE}`
 
   require_relative 'lib/asciidoctor-revealjs'
 end

